### DISCUSSION TITLE
Continue pretraining

### DISCUSSION THREAD

**Discussion by HunterKruger** [s-JoL/Open-Llama#58] [General]
Hello, sometimes a pretrain train breaks, we have to continue pretraining. The latest version support this, but how to skip the zst files which have already been fed into the model?

**Comment by s-JoL** [comment]
At present, the function of skipping used samples is realized by fixing the order of samples in datasets and the skip_first_batches function provided by accelerate. See https://github.com/s-JoL/Open-Llama/blob/main/solver/trainer.py#L113 for the specific implementation.

**Reply by HunterKruger** [reply]
Really appreciated !
